apps/web/src/stores/experiments.ts

```diff
@@ -107,34 +107,40 @@ export function useExperiments(
   const { execute: create, isPending: isCreating } = useLatitudeAction(
     createExperimentAction,
     {
-      onSuccess: async ({ data: experiment }: { data: Experiment }) => {
-        mutateCount((prev) => (prev ? prev + 1 : 1), {
+      onSuccess: async ({ data: experiments }: { data: Experiment[] }) => {
+        mutateCount((prev) => (prev ?? 0) + experiments.length, {
           revalidate: false,
         })
 
-        const experimentDto = {
-          ...experiment,
-          results: {
-            passed: 0,
-            failed: 0,
-            errors: 0,
-            totalScore: 0,
-          },
-        } as ExperimentDto
+        const experimentDtos = experiments.map(
+          (experiment) =>
+            ({
+              ...experiment,
+              results: {
+                passed: 0,
+                failed: 0,
+                errors: 0,
+                totalScore: 0,
+              },
+            }) as ExperimentDto,
+        )
 
         if (page === 1) {
           mutate(
             (prev) => {
-              const prevExperiment = prev?.find(
-                (exp) => exp.uuid === experimentDto.uuid,
-              )
-
-              if (prevExperiment) {
-                // this might happen due to race conditions with the experimentStatus websocket
```

Comment:
hmm i know this ain't you but the store should not know anything about websocket shenanigans, it's a smell